### ISSUE TITLE
Ragnarok: Path of Heroes

### ISSUE THREAD

**Issue by txtsd** [devadvance/rootcloak#98]
This app detects root no matter what

https://play.google.com/store/apps/details?id=com.us.ragnarok

**Comment by pylerSM** [comment]
"no matter what" - what did you try?

**Comment by txtsd** [comment]
@pylerSM I've added the app in `Manage Apps` and even tried disabling `Native Root Detection` for the app, but on launching Ragnarok Online the app says it won't run on rooted phones.

**Comment by hikaritenchi** [comment]
A logcat might be helpful to diagnose.

**Comment by txtsd** [comment]
@hikaritenchi towards the end of the file
[logcat.txt](https://github.com/devadvance/rootcloak/files/393188/logcat.txt)

**Comment by hikaritenchi** [comment]
Looks like a potential problem with com.chelpus.XSupport, but I'm not familiar with that module. Can you try disabling it and trying again? (and if it doesn't work, another logcat)

**Comment by txtsd** [comment]
@hikaritenchi Here's the log without that app and Xposed module installed: 
[logcat.txt](https://github.com/devadvance/rootcloak/files/400196/logcat.txt)

**Comment by Spodi** [comment]
Same problem happens with summoner war (also from com2us), should use the same detection. My phine produces a hell of logcat spam, so i dont know if it would help...

**Comment by pylerSM** [comment]
Lucky Patcher hooks this apk "com.us.ragnarok" but in hook he somehow checks for superuser.apk file in "code of this app" .. Disable Lucky Patcher hooks for this app (if possible) or whole module until LuckyPatcher dev handle it yourself. He should really check if file exists, before he tries to open it (java.io.FileNotFoundException: /system/app/FAKEJUNKFILE.apk: open failed: ENOENT (No such file or directory)
08-01 13:18:48.314 W/System.err(15344): 	at libcore.io.IoBridge.open(IoBridge.java:452))

Try & tell us results.

**Comment by txtsd** [comment]
@pylerSM The second logcat is without Lucky Patcher installed and without its Xposed module.

`/system/app/FAKEJUNKFILE.apk` does not exist. What even is that?

**Comment by hikaritenchi** [comment]
Your second logcat doesn't have any RootCloak xposed logs, unfortunately...Remember to verify that the logs are enabled before trying to capture a logcat, thanks!

**Comment by pylerSM** [comment]
/system/app/FAKEJUNKFILE.apk is fake path instead of /system/app/Superuser.apk.. so file not found..

**Comment by txtsd** [comment]
@hikaritenchi @pylerSM Does this file have the needed information?
[logcat.txt](https://github.com/devadvance/rootcloak/files/405687/logcat.txt)

**Comment by hikaritenchi** [comment]
No, sorry. Verify that there are lines similar to the ones below:

```
08-08 16:28:42.550 ? I/Xposed: Loaded app: klb.android.lovelive
08-08 16:28:42.553 ? I/Xposed: No need to change build tags: release-keys
08-08 16:28:42.712 ? I/Xposed: File: Found a File constructor: /data/app/klb.android.lovelive-2/base.apk
08-08 16:28:42.762 ? I/Xposed: Hooked loadLibrary
08-08 16:28:42.762 ? I/Xposed: File: Found a File constructor: /data/app/klb.android.lovelive-2/lib/arm, with: libjniproxy.so
08-08 16:28:42.793 ? I/Xposed: File: Found a File constructor: /data/user/0/klb.android.lovelive, with: shared_prefs
```
